runtime/parachains/src/paras_inherent/mod.rs

```diff
@@ -489,14 +483,15 @@ impl<T: Config> Pallet<T> {
 		}
 
 		METRICS.on_candidates_included(freed_concluded.len() as u64);
-		let freed = collect_all_freed_cores::<T, _>(freed_concluded.iter().cloned());
-
-		<scheduler::Pallet<T>>::clear();
-		<scheduler::Pallet<T>>::schedule(freed, now);
+		let mut freed = collect_all_freed_cores::<T, _>(freed_concluded.iter().cloned());
+		// As a core is only occupied by a single work item, freed and freed_disputed are disjoint by design
```

Comment:
Either cores are freed due to disputes (`freed_disputed`) or freed due to timeout or conclusion (`freed`) (not my names). I don't believe the same core can be freed due to more than one reason. Hence, these two collections are disjoint since a core is either freed due to disputes, or timeout, or conclusion.

Reply:
I don't think this holds. There is no reason a core would not be able to time out and its candidate get disputed at the same time. At least I can't think of a reason, why this would not be possible.

Reply:
Fair enough. This needs to change.